target_postgres/json_schema.py

```diff
@@ -1,12 +1,186 @@
+import re
 
-def sanitize_type(type):
+
+def get_type(schema):
     """
-    Given a potentially non-array type instance, return an array type instance
-    :param type: str, array
-    :return: [type ...]
+    Given a JSON Schema dict, extracts the simplified `type` value
+    :param schema: dict, JSON Schema
+    :return: [string ...]
     """
+    type = schema.get('type', None)
+    if not type:
+        return ['object']
 
     if isinstance(type, str):
         return [type]
 
     return type
+
+
+def _get_ref(schema, paths):
+    if not paths:
+        return schema
+
+    return _get_ref(schema[paths[0]], paths[1:])
+
+
+def get_ref(schema, ref):
+    """
+    Given a JSON Schema dict, and a valid ref (`$ref`), get the JSON Schema from within schema
+    :param schema: dict, JSON Schema
+    :param ref: string
+    :return: dict, JSON Schema
+    """
+
+    return _get_ref(schema,
```

Comment:
Should `id` based refs be supported? https://json-schema.org/understanding-json-schema/structuring.html#using-id-with-ref